errors.go

```diff
@@ -106,6 +106,17 @@ func (e ConflictError) StatusCode() int { return http.StatusConflict }
 
 func (e ConflictError) Unwrap() error { return HTTPError(e) }
 
+// InternalServerError is an error used to return a 500 status code.
+type InternalServerError HTTPError
+
+var _ ErrorWithStatus = InternalServerError{}
+
+func (e InternalServerError) Error() string { return e.Err.Error() }
+
+func (e InternalServerError) StatusCode() int { return http.StatusInternalServerError }
+
+func (e InternalServerError) Unwrap() error { return HTTPError(e) }
```

Comment:
HTTPError supports internal error in StatusCode, so an httperror is directly an internal error

But by looking at the implementation I think the issue the fact HTTPError.Error should be modified to call HTTPError.StatusCode() and not HTTPError.Code

```diff
func (e HTTPError) Error() string {
        title := e.Title
+       code := e.StatusCode()
        if title == "" {
-                title = http.StatusText(e.Status)
+                title = http.StatusText(code)
                if title == "" {
                        title = "HTTP Error"
                }
        }
-        return fmt.Sprintf("%s (%d): %s", title, e.Status, e.Detail)
+        return fmt.Sprintf("%s (%d): %s", title, code, e.Detail)
}
```